modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -40,10 +139,59 @@ static void boundingRectTestBody(const In& in, const CompareRects& cmpF, cv::GCo
 {
     cv::Rect out_rect_gapi;
     boundingRectTestGAPI(in, std::move(args), out_rect_gapi);
-
     boundingRectTestOpenCVCompare(in, out_rect_gapi, cmpF);
 }
 
+//-------------------------------------------------------------------------------------------------
+
+template<typename In>
+static cv::GComputation fitLineTestGAPI(const In& in, const cv::DistanceTypes distType,
+                                        cv::GCompileArgs&& args, cv::Vec4f& out_vec_gapi)
+{
+    double paramDefault = 0., repsDefault = 0., aepsDefault = 0.;
+
+    cv::detail::g_type_of_t<In> g_in;
+    auto out = cv::gapi::fitLine2D(g_in, distType, paramDefault, repsDefault, aepsDefault);
+    cv::GComputation c(cv::GIn(g_in), cv::GOut(out));
+    c.apply(cv::gin(in), cv::gout(out_vec_gapi), std::move(args));
+    return c;
+}
+
+template<typename In>
+static cv::GComputation fitLineTestGAPI(const In& in, const cv::DistanceTypes distType,
```

Comment:
I see difference in fitLine_D function only. Otherwise, these functions (fitLineTestGAPI) are identical. Can you shorten code?

Reply:
Despite the fact the difference seems to be only in one string, all the code after `auto out = cv::gapi::fitLine3D` is linked to this string to match the types. So, in one function it would look like that:
```
cv::detail::g_type_of_t<In> g_in;
if (dim == 4)
{
    auto out = cv::gapi::fitLine2D(g_in, distType, paramDefault, repsDefault, aepsDefault);
    cv::GComputation c(cv::GIn(g_in), cv::GOut(out));
    c.apply(cv::gin(in), cv::gout(out_vec_gapi), std::move(args));
    return c;
}
}
else
{
    auto out = cv::gapi::fitLine3D(g_in, distType, paramDefault, repsDefault, aepsDefault);
    cv::GComputation c(cv::GIn(g_in), cv::GOut(out));
    c.apply(cv::gin(in), cv::gout(out_vec_gapi), std::move(args));
    return c;
}
```
So, all the code will be split to two blocks. no profit. I don't think it is better than just other overload/template specialization.